src/readers/morphologyASC.cpp

```diff
@@ -139,9 +140,24 @@ class NeurolucidaParser
             nb_.addMarker(marker);
             return_id = -1;
         } else if (header.token == Token::CELLBODY) {
-            if (!nb_.soma()->points().empty())
-                throw SomaError(err_.ERROR_SOMA_ALREADY_DEFINED(lex_.line_num()));
-            nb_.soma()->properties() = properties;
+            if (!nb_.soma()->points().empty()) {
+                if (options & ALLOW_SOMA_BIFURCATIONS) {
+                    printError(Warning::SOMA_BIFURCATION, err_.WARNING_SOMA_BIFURCATION());
+                } else if (options & ALLOW_MULTIPLE_SOMATA) {
+                    printError(Warning::MULTIPLE_SOMATA, err_.WARNING_MULTIPLE_SOMATA());
+                } else {
+                    throw SomaError(err_.ERROR_SOMA_ALREADY_DEFINED(lex_.line_num()));
+                }
+                nb_.soma()->properties()._points.insert(nb_.soma()->properties()._points.end(),
```

Comment:
There is no further distinction between bifurcation and multiple soma? They just get merged into one?

Reply:
For the ASC file format they appeared to be processed the same way. But for SWC it's different.

Reply:
Ah, thanks for elaborating!